kernel/src/table.rs

```diff
@@ -106,12 +112,12 @@ fn resolve_uri_type(table_uri: impl AsRef<str>) -> DeltaResult<UriType> {
         } else if scheme.len() == 1 {
             // NOTE this check is required to support absolute windows paths which may properly
             // parse as url we assume here that a single character scheme is a windows drive letter
-            Ok(UriType::LocalPath(PathBuf::from(table_uri)))
+            Ok(UriType::LocalPath(PathBuf::from(table_uri.as_ref())))
         } else {
             Ok(UriType::Url(url))
         }
     } else {
-        Ok(UriType::LocalPath(table_uri.into()))
+        Ok(UriType::LocalPath(PathBuf::from(table_uri.as_ref())))
```

Comment:
Out of curiosity, does this not work?
```suggestion
        Ok(UriType::LocalPath(PathBuf::from(&table_uri)))
```
(perhaps the `Deref` from `Cow` to `&str` followed by `AsRef` from &str to `&OsStr` is too much indirection?)

Reply:
Tho thinking more, this is probably better anyway, if it works?
```suggestion
        Ok(UriType::LocalPath(table_uri.deref().into()))
```